#                             Parkinson - Disease -Classifier

## Let Us Import some libraries



In [19]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
from sklearn import metrics
from sklearn.linear_model import LogisticRegression


#### Data
Data is read from file which was created from dat_file.py using data from the dataset

In [20]:
data = pd.read_csv("file.csv")


In [21]:
data

pressure  grip_angle    timestamp    state  status
0    884.200931  794.317300   115.650000   static       1
1    910.966791  662.641217   115.650000  dynamic       1
2    788.450502  782.556671  1052.316667   static       1
3    926.910511  902.801802  1052.316667  dynamic       1
4    843.550013  903.502152  1228.116667   static       1
..          ...         ...          ...      ...     ...
149  472.697735  940.653476   149.566667  dynamic       0
150  532.731178  603.585907   268.833333   static       0
151  593.152854  639.329249   268.833333  dynamic       0
152  532.731178  603.585907   268.833333   static       0
153  593.152854  639.329249   268.833333  dynamic       0

[154 rows x 5 columns]

Let us look at the missing data

In [22]:
print(data.isnull().sum())

pressure      1
grip_angle    1
timestamp     0
state         0
status        0
dtype: int64


###### we have missing data in pressure and grip angle lets remove that row alone

In [23]:
data = data.dropna() 

lets look at the data

In [25]:
data

pressure  grip_angle    timestamp    state  status
0    884.200931  794.317300   115.650000   static       1
1    910.966791  662.641217   115.650000  dynamic       1
2    788.450502  782.556671  1052.316667   static       1
3    926.910511  902.801802  1052.316667  dynamic       1
4    843.550013  903.502152  1228.116667   static       1
..          ...         ...          ...      ...     ...
149  472.697735  940.653476   149.566667  dynamic       0
150  532.731178  603.585907   268.833333   static       0
151  593.152854  639.329249   268.833333  dynamic       0
152  532.731178  603.585907   268.833333   static       0
153  593.152854  639.329249   268.833333  dynamic       0

[153 rows x 5 columns]

* we have removed the row which contains null values,let's check again 

In [26]:
print(data.isnull().sum())

pressure      0
grip_angle    0
timestamp     0
state         0
status        0
dtype: int64


lets split the data using in built functions from the sklearn lib 

In [27]:
# Let us now split the data into train and test before it shuffle the data

X_train, X_test, y_train, y_test = train_test_split(data, data.status,test_size=0.2, random_state=42, shuffle=True)


##### Print the shapes of training and test datasets

In [28]:
print(X_train.shape)
print(y_train.shape)
print(X_test.shape)
print(y_test.shape)


(122, 5)
(122,)
(31, 5)
(31,)


## Random Forest Classifier
 ###### Let's add the 4 Features from the file  and train our  model  

In [29]:
'''
RANDOM FOREST MODEL FOR CLASSIFICATION
'''

features = ["pressure","grip_angle","timestamp","state"]

X = pd.get_dummies(X_train[features])
X_test = pd.get_dummies(X_test[features])

model = RandomForestClassifier(n_estimators = 50,max_depth = 4 ,random_state=0 ) 
model.fit(X,y_train)
y_pred = model.predict(X_test)
print("Accuracy of Random Forest:",metrics.accuracy_score(y_test, y_pred))


Accuracy of Random Forest: 0.967741935483871


Bingo! we have got an accuracy of 0.967741935483871
Tune the hyperparameters to get more accuracy

# Logistic Regression

In [30]:
'''
LOGISTIC REGRESSION MODEL FOR CLASSIFICATION
'''
model2 = LogisticRegression(penalty= 'none' ,random_state=0 ,max_iter=100,solver='lbfgs').fit(X,y_train)
y_pred2 = model2.predict(X_test)

print("Accuracy of Logistic regression:",metrics.accuracy_score(y_test, y_pred2))

Accuracy of Logistic regression: 0.9032258064516129


Note: This file is still to be updated